vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -403,8 +404,10 @@ func (c completedConfig) New(name string, delegationTarget DelegationTarget) (*G
 
 		listedPathProvider: apiServerHandler,
 
-		swaggerConfig: c.SwaggerConfig,
-		openAPIConfig: c.OpenAPIConfig,
+		swaggerConfig:        c.SwaggerConfig,
+		openAPIConfig:        c.OpenAPIConfig,
+		swaggerAPIContainers: append([]*restful.Container{apiServerHandler.GoRestfulContainer}, delegationTarget.SwaggerAPIContainers()...),
```

Comment:
> why do you cache this list in the server? Can't you build up the list when you need it?

Build it up from what?  The delegate is "lost" after construction to avoid side-effecty references without planning/structure.



> Why do we need the list in the first place? Because apiServerHandler.GoRestfulContainer can be nil? Would it be enough to have the first non-nil container?

No.  We have to make a tweak to the upstream swagger generator to allow multiple containers to be unioned to produce the swagger that we're looking for.  They are *not* all unioned in a single gorestful container.  They delegate.